src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1011,6 +995,7 @@ private Task WriteSuffix()
 
             if (!HasFlushedHeaders)
             {
+                _requestProcessingStatus = RequestProcessingStatus.HeadersFlushed;
                 return FlushAsyncInternal(default).AsTask();
```

Comment:
Remove FlushAsyncInternal entirely and just call Output.FlushAsync(ct);

Reply:
Done for all the other places; changed this to 
```csharp
return FlushAsyncInternal();
```
Then had `FlushAsyncInternal` do the `.AsTask()` since its no-inline so it doesn't pull the `ValueTask<FlushResult>` into this stackframe as it should rarely be called e.g.
```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
private Task FlushAsyncInternal()
{
    return Output.FlushAsync(default).AsTask();
}
```